src/app/technical-record/technical-record.component.ts

```diff
@@ -85,7 +94,7 @@ export class TechnicalRecordComponent implements OnInit {
   }[]>;
   tc3Inspections$: Observable<any[]>;
 
-  constructor(private _store: Store<IAppState>, public matDialog: MatDialog) {
+  constructor(private _store: Store<IAppState>, public matDialog: MatDialog, public router: Router) {
```

Comment:
we need to use ngrx-router state to tap router actions in effects , that way we can retain the SOLID design's first paradigm , which is separation of concern.

component is related with UI (View) rendering only , while we use effects to respond to changes due to user actions

Reply:
please can you remove angular router injection in the component